src/tir/transforms/storage_flatten.cc

```diff
@@ -1357,7 +1427,8 @@ class StorageFlattener : public StmtExprMutator {
       }
 
       e.buffer = Buffer(op->buffer->data, op->buffer->dtype, op->buffer->shape, op->buffer->strides,
-                        PrimExpr(), op->buffer->name, align, 0, kDefault);
+                        PrimExpr(), op->buffer->name, align, 0, kDefault,
+                        op->buffer->axis_separators, op->buffer->span);
```

Comment:
Were the StorageFlatten changes missed in #9727 and being added here now that we are integrating with codegen and your test caught an issue? 

I was initially confused why storage flatten needs a change here but ⬆️   is my guess now. If not can you comment on the issue? Thanks

Reply:
That's correct, the changes to storage flatten are bugfixes that were exposed by these additional tests at the codegen level.  Here, the `axis_separators` failed to be passed through to the buffer with alignment, and earlier in the file the `IndexMap` failed to be updated when the buffer objects were modified.